### PR TITLE
Infer first and last name from github and twitter response

### PR DESCRIPTION
**Pull request by igorauad**
I think `firstName` and `lastName` should be set when signing through providers. This fixes it for github and twitter.

### CODE REVIEW

**Comment by ilanbiala** [comment]
@igorauad once you make the changes I'll review again and merge if it is good.

**Comment by igorauad** [comment]
Thanks, @ilanbiala . It supports now last name with multiple names, as you suggested.

File: config/strategies/github.js [hunk 1]
```diff
@@ -23,8 +23,14 @@ module.exports = function() {
 			providerData.refreshToken = refreshToken;
 
 			// Create the user OAuth profile
+			var displayName = profile.displayName.trim();
+			var firstName =  displayName.indexOf(' ') !== -1 ? displayName.substring(0, displayName.indexOf(' ')) : displayName;
+			var lastName = displayName.indexOf(' ') !== -1 ? displayName.substring(displayName.indexOf(' ') + 1) : undefined;
```
Comment (ilanbiala) [inline config/strategies/github.js]: Since we're using indexOf more than once, why don't we cache that value in a variable?

**Comment by ilanbiala** [comment]
Just one change to save on indexOf calls, I think for larger names it's more useful, but still 4 down to 1.

File: config/strategies/github.js (hunk 1 above)
Comment (igorauad) [inline config/strategies/github.js]: You are absolutely right.

**Comment by ilanbiala** [comment]
@igorauad looks great, can you squash the commits and reword the last commit? If you don't know how you can refer to this article http://davidwalsh.name/squash-commits-git.

**Comment by ilanbiala** [comment]
LGTM, @lirantal any objections?

**Comment by lirantal** [comment]
for lastName, isn't it preferred that we set to an empty string '', instead of undefined?

**Comment by ilanbiala** [comment]
@lirantal yeah that's clearer and probably more compatible actually. @igorauad can you make the change @lirantal suggested and squash it to 1 commit again?

**Comment by igorauad** [comment]
Sure, @ilanbiala @lirantal

**Comment by ilanbiala** [comment]
Thanks @igorauad for the good PR. Keep 'em coming.

**Comment by igorauad** [comment]
Thank you, @ilanbiala !

**Pull request merged**